src/register_target.cpp

```diff
@@ -57,8 +58,25 @@ target make_target(const std::string& name)
 {
     if(not contains(target_map(), name))
     {
-        auto target_name = make_shared_object_filename("migraphx_" + name);
-        store_target_lib(dynamic_loader(target_name));
+        std::string so_version = "." + std::to_string(MIGRAPHX_SO_MAJOR_VERSION) + ".0";
+        auto target_name       = make_shared_object_filename("migraphx_" + name);
+
+        // Try to load library with so version in the name.
+        // If library with so version name is not found,
+        // try loading the library without the so version name.
+        // For example, if "libmigraphx_ref.so.2010000.0" is not found,
+        // try loading "libmigraphx_ref.so".
+        try
+        {
+            // Default to loading shared libraries with so version in their name.
+            store_target_lib(dynamic_loader(target_name + so_version));
+        }
+        catch(...)
+        {
+            // Load the library without the so version in the name.
+            std::cout << "In catch block..." << std::endl;
```

Comment:
Remove this `cout`

Reply:
Good catch! I missed to remove my testing instructions. Thanks!